File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    
    int i = 0 ;
    int prev = -1;
    int size = arr.size();
    
    while(i < size){
        
        int now = arr.at(i);
        
        if(prev != now) {
            answer.push_back(arr[i]);
            prev = arr[i];
        }
        
        i++;
    }

    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    index = len(arr)
    
    if index == 1:
        answer.append(arr[0])
        return answer
    
    else:
        for i in range(index-1):
            if arr[i] != arr[i+1]:
                answer.append(arr[i])

        if answer and arr[-1] != answer[-1] and answer:
            answer.append(arr[-1])     
        else: 
            answer.append(arr[0])

            

    
    return answer